src/v/coproc/tests/supervisor.h

```diff
@@ -33,18 +35,18 @@ class supervisor final : public coproc::supervisor_service {
       : supervisor_service(sc, ssg)
       , _coprocessors(cp_map) {}
 
-    ~supervisor() { _gate.close().get(); }
+    ~supervisor() override { _gate.close().get(); }
```

Comment:
make a ss::future<> stop() and close the gate in there

Reply:
Its working as of now, just checked this is a more involved change, saving for later